usr/lib/ica_s390_stdll/ica_specific.c

```diff
@@ -3589,8 +3589,45 @@ static CK_BBOOL isMechanismHW(STDLL_TokData_t *tokdata, CK_ULONG mechanism)
     return FALSE;
 }
 
+#define KEY_SIZE_512   0x01
+#define KEY_SIZE_1024  0x02
+#define KEY_SIZE_2048  0x04
+#define KEY_SIZE_4096  0x08
+
+static void adjust_key_sizes(unsigned int mask, CK_MECHANISM_INFO *mech_info)
+{
+    CK_ULONG min = 0, max = 0;
+
+    if (mask == 0)
+        return;
+
+    if (mask & KEY_SIZE_512)
```

Comment:
simpler code would be `min = ((int)mask & -(int)mask) * 512`, but only if the "isolate right-most one" pattern is known and accepted.

Reply:
Hm well, since there is no simple "isolate left-most one" pattern, I would tend to leave both as is....